docs/sources/reference/components/otelcol/otelcol.receiver.syslog.md

```diff
@@ -0,0 +1,271 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.receiver.syslog/
+description: Learn about otelcol.receiver.syslog
+title: otelcol.receiver.syslog
+---
+
+# otelcol.receiver.syslog
+
+`otelcol.receiver.syslog` accepts syslog messages over the network and forwards them as logs to other `otelcol.*` components.
+It supports syslog protocols [RFC5424][] and [RFC3164][] and can receive data over `TCP` or `UDP`.
+
+{{< admonition type="note" >}}
+`otelcol.receiver.syslog` is a wrapper over the upstream OpenTelemetry Collector `syslog` receiver.
+Bug reports or feature requests will be redirected to the upstream repository, if necessary.
+{{< /admonition >}}
+
+You can specify multiple `otelcol.receiver.syslog` components by giving them different labels.
+
+[RFC5424]: https://www.rfc-editor.org/rfc/rfc5424
+[RFC3164]: https://www.rfc-editor.org/rfc/rfc3164
+
+## Usage
+
+```alloy
+otelcol.receiver.syslog "LABEL" {
+  tcp { ... }
+  udp { ... }
+
+  output {
+    logs    = [...]
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+| Name                              | Type     | Description                                                        | Default   | Required |
+|-----------------------------------|----------|--------------------------------------------------------------------|-----------|----------|
+| `protocol`                        | `string` | The syslog protocol that the syslog server supports.               | `rfc5424` | no       |
+| `location`                        | `string` | The geographic time zone to use when parsing an RFC3164 timestamp. | `UTC`     | no       |
+| `enable_octet_counting`           | `bool`   | Whether to enable RFC6587 octet counting.                          | `false`   | no       |
+| `max_octets`                      | `int`    | The maximum octets for messages when octet counting is enabled.    | `8192`    | no       |
+| `allow_skip_pri_header`           | `bool`   | Allow parsing records without a priority header.                   | `false`   | no       |
+| `non_transparent_framing_trailer` | `string` | The framing trailer when using RFC6587 Non-Transparent-Framing.    | `nil`     | no       |
+
+The `protocol` argument specifies the syslog format supported by the receiver.
+`protocol` must be one of `rfc5424`, `rfc3164`
+
+The `location` argument specifies a Time Zone identifier. The available locations depend on the local IANA Time Zone database.
+See [this wikipedia entry][tz-wiki] for a non-comprehensive list.
+
+The `non_transparent_framing_trailer` argument must be one of `LF`, `NUL`.
```

Comment:
Can we add some context around this? Like how can the default be nil? Even if its a string it should be "" but that seems to not be LF or NUL.

Reply:
Yeah, that's fair. It's a *string, so it's nil if not set. If it is set, then the character chosen is expected to terminate the message. If not set, it's expected that you're using octet counting or that the syslog frame contains only one message.